src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1147,6 +1205,17 @@ private AuthenticationResult acquireTokenAfterValidation(CallbackHandler callbac
             AuthenticationResult result = null;
             request.setVersion(getVersionName());
             request.setBrokerAccountName(request.getLoginHint());
+            
+            //check if App gives permission to AccountManager
+            try
```

Comment:
This is why I asked if we need to refactor the canSwitchToBroker() function on Slack disc channel in the first place. And you said we should keep it in the canSwitchToBroker and in the same time check the permissions after going to the broker flow. This also confused me.